tests/spread-function-call/spread-function-call.es5

```diff
@@ -0,0 +1,47 @@
+var list = [1, 2, 3];
+
+Array.prototype.expand = false;
+
+function __(list) {
+  list.expand = true;
+  return list;
+}
+
+// wrap the function to be called, so that the arguments are parsed first.
+function wrap(func) {
```

Comment:
I'm wondering if it's more representative to have specific instances then. Vs. a generic solution that has a lot of overhead in complexity and runtime (args walking, etc).

I.e.

```
function fn(a, b, c) {
  return c;
}

fn();
fn.apply(null, list);
fn.apply(null, [3, [2, 1, 0]].concat(list));
```